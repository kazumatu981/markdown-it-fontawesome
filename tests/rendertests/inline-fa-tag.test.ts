import { describe, test, expect } from '@jest/globals';
import MarkdownIt from 'markdown-it';
import { InlineFaTagTest } from './snapshot-testcases';

const markdownItFontawesome = require('../../index');

function render(markdown: string): string {
    const md = new MarkdownIt();
    md.use(markdownItFontawesome);

    return md.render(markdown);
}

describe('Inline FaTag Tests.', () => {
    InlineFaTagTest.forEach((testCase) => {
        test(testCase.description, () => {
            const actual = render(testCase.markdown);
            expect(actual).toMatchSnapshot();
        });
    });
});
