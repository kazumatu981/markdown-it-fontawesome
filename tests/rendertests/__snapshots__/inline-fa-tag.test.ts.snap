// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Inline FaTag Tests. simple tag. ':fa fa-user:' 1`] = `
"<p><i class="fa fa-user"></i></p>
"
`;

exports[`Inline FaTag Tests. stacking simple tag. '[:fa fa-yyy: :fa fa-xxx:]' 1`] = `
"<p><span class="fa-stack"><i class="fa fa-yyy"></i><i class="fa fa-xxx"></i></span></p>
"
`;

exports[`Inline FaTag Tests. stacking styled tag. '[:fa fa-yyy:{.red} :fa fa-xxx:{.blue}]' 1`] = `
"<p><span class="fa-stack"><span class="red"><i class="fa fa-yyy"></i></span><span class="blue"><i class="fa fa-xxx"></i></span></span></p>
"
`;

exports[`Inline FaTag Tests. styled tag (malti style). ':fa fa-user:{.red .big}' 1`] = `
"<p><span class="red big"><i class="fa fa-user"></i></span></p>
"
`;

exports[`Inline FaTag Tests. styled tag. ':fa fa-user:{.red}' 1`] = `
"<p><span class="red"><i class="fa fa-user"></i></span></p>
"
`;

exports[`Inline FaTag Tests. very simple tag. ':fa fa-user:' 1`] = `
"<p><i class="fa-user"></i></p>
"
`;
